src/File.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace React\Http;
+
+use React\Stream\ReadableStreamInterface;
+
+class File implements FileInterface
+{
+    /**
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * @var string
+     */
+    protected $filename;
+
+    /**
+     * @var string
+     */
+    protected $type;
+
+    /**
+     * @var ReadableStreamInterface
+     */
+    protected $stream;
+
+    /**
+     * @param string $name
+     * @param string $filename
+     * @param string $type
+     * @param ReadableStreamInterface $stream
+     */
+    public function __construct($name, $filename, $type, ReadableStreamInterface $stream)
+    {
+        $this->name = $name;
+        $this->filename = $filename;
+        $this->type = $type;
+        $this->stream = $stream;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
```

Comment:
Perhaps my previous remark was a bit unclear, "name" is certainly essential here because individual fields need to be identified :)

If you look at PHP's `$_FILES['uploaded_file']` structure, you'll see the following fields:

* `name` maps to our `getFilename()`
* `type` maps to our `getType()`
* `size` does not apply here because we're streaming the file contents
* `tmp_name` does not apply either
* `error` does not apply either

Noticeably, the "field name" is not part of the structure (which does make sense IMHO). This raises the question why we should add this as part of the `FileInterface`?

(Note I'm not saying there are no valid reasons for either decision, I merely want this to be deliberate and with proper documentation)